src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Utilities/ConfiguredProjectExtensions.cs

```diff
@@ -6,6 +6,12 @@ namespace Microsoft.VisualStudio.ProjectSystem;
 
 internal static class ConfiguredProjectExtensions
 {
+    public static string GetProjectName(this ConfiguredProject project)
+        => GetProjectName(project.UnconfiguredProject);
+
+    public static string GetProjectName(this UnconfiguredProject project)
+        => Path.GetFileNameWithoutExtension(project.FullPath);
```

Comment:
Seems like an opportunity to use an extension property, if you wanted to add one.

Reply:
Though CPS should really add this capability so that we don't allocate every time this is called. We did something similar for the project directory: https://dev.azure.com/devdiv/DevDiv/_git/CPS/pullrequest/594043

Reply:
https://dev.azure.com/devdiv/DevDiv/_git/CPS/pullrequest/670939